src/read_write/s2.rs

```diff
@@ -0,0 +1,76 @@
+use crate::read_write::{Encoding, NodeWriter};
+use crate::s2_geo::{cell_id, ECEFExt};
+use crate::{AttributeData, PointsBatch};
+use std::collections::{BTreeMap, HashMap};
+use std::io::Result;
+use std::path::PathBuf;
+
+pub struct S2Splitter<W> {
+    writers: HashMap<String, W>,
+    stem: PathBuf,
+    split_level: u8,
+}
+
+impl<W> S2Splitter<W>
+where
+    W: NodeWriter<PointsBatch>,
+{
+    pub fn new(path: impl Into<PathBuf>, split_level: u8) -> Self {
+        let writers = HashMap::new();
+        let stem = path.into();
+        S2Splitter {
+            writers,
+            stem,
+            split_level,
+        }
+    }
+
+    pub fn write(&mut self, points_batch: &PointsBatch) -> Result<()> {
+        let mut out = HashMap::new();
```

Comment:
Nit: Consider naming this `batches_by_s2cell` and/or adding a comment that this is a buffer with one entry per S2Cell.